app/assets/javascripts/leaflet_helper.js

```diff
@@ -0,0 +1,55 @@
+  function setupLeafletMap() {
+    //Bounding map.  
+    var bounds = new L.LatLngBounds(new L.LatLng(84.67351257 , -172.96875) , new L.LatLng(-54.36775852 , 178.59375)) ;
+    var map = L.map('map_leaflet' , {
+      maxBounds: bounds , 
+      maxBoundsViscosity: 0.75
+    }) ;
+    return map ;
+  }
+
+  function setupLayers(map) {
+    var mapboxUrl = "//a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png" ;
+    var normal_layer = L.tileLayer(mapboxUrl, {id: 'map'}) ; 
+    normal_layer.addTo(map) ; 
+    map.options.minZoom = 1 ;
+    var baseMaps = {
+      "Default": normal_layer,
+    };
+    var overlayMaps = {
+      "Skynet": layerGroup    // we can add more layers here !
+    }; 
+    L.control.layers(baseMaps , overlayMaps).addTo(map);
+  }
+
+  function setupFullScreen(map , lat , lon) {
```

Comment:
Really liked the way you removed the script from view and added and divided in functions in separate file :100:

Reply:
Thank you @Gauravano 😄 !